components/FilterComponent.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <div class="text-center">
+    <div v-if="!filterData.active">
+      <b-row class="mb-3">
+        <b-col cols="3" class="mt-2">
+          Only display runs with a
+        </b-col>
+        <b-col cols="7">
+          <multiselect v-model="filterData.details" :options="filterLabels" track-by="index" label="label"/>
+        </b-col>
+        <b-col cols="2" class="mt-2">
+          time between
+        </b-col>
+      </b-row>
+      <b-row class="mt-3">
+        <b-col cols="5">
+          <time-selector v-model="filterData.timeMin"/>
+        </b-col>
+        <b-col cols="2" class="mt-2">
+          and
+        </b-col>
+        <b-col cols="5">
+          <time-selector v-model="filterData.timeMax"/>
+        </b-col>
+      </b-row>
+      <div v-b-tooltip.hover :title="allowActivate">
+        <b-button @click="activateFilter" variant="success" class="mt-2" :disabled="allowActivate != ''">
+          Activate filter
+        </b-button>
+      </div>
+    </div>
+    <div v-else>
+      <b-row>
+        <b-col cols="9" class="mt-auto">
+          <h5>{{ filterDescription }}</h5>
+        </b-col>
+        <b-col cols="3">
+          <b-button @click="deleteFilter" variant="danger" class="mt-2">Delete filter</b-button>
+        </b-col>
+      </b-row>
+    </div>
+    <hr>
+  </div>
+</template>
+
+<script lang="ts">
+import {Component, Prop, Vue}                          from 'nuxt-property-decorator';
+import {SplitFile, selectTime}                         from '~/util/splits';
+import {stringTimeToSeconds, secondsToFormattedString} from '~/util/durations';
+import {Filter}                                        from '~/util/filter';
+import store                                           from '~/util/store';
+import Multiselect                                     from 'vue-multiselect';
+
+@Component({components: {Multiselect}})
+export default class FilterComponent extends Vue {
+  @Prop()
+  filterIndex!: number;
+
+  @Prop()
+  parsedSplits!: SplitFile;
+
+  useRealTime: boolean = store.state.useRealTime;
+
+  globalFilters: Filter[] = store.state.filters;
+
+  filterData: Filter = {details: {label: "", index: -2}, timeMin: 0, timeMax: 0, active: false, attempts: []};
+
+  get filterLabels() {
+    let options = [{index: -1, label: "Global"}];
+    this.parsedSplits.Run.Segments.Segment.forEach((split, index) => options.push({index: index, label: split.Name}));
+
+    return options;
+  }
+
+  get allowActivate() {
+    if (this.filterData.details?.index == -2 || this.filterData.details?.index == undefined)
```

Comment:
I see this `-2` a lot, it should probably be a named constant somewhere (in filter.ts probably), same for `-1` to indicate "Global"